pilot/pkg/networking/core/configgen.go

```diff
@@ -35,7 +35,7 @@ type ConfigGenerator interface {
 	BuildListeners(node *model.Proxy, push *model.PushContext) []*listener.Listener
 
 	// BuildClusters returns the list of clusters for the given proxy. This is the CDS output
-	BuildClusters(node *model.Proxy, push *model.PushContext) ([]*discovery.Resource, model.XdsLogDetails)
```

Comment:
Having to pass boolean to detect delta or not to BuildClusters does not seem right in the interface. Can we add additional method for delta like BuildDeltaClusters that takes additional updates and cdsGenerator can call that when it is delta mode?

Reply:
In the implementation we can reuse with an internal method.

Reply:
I think adding a delta flag would involve less changes -- and is just simpler overall. `BuildDeltaClusters` would require no-op implementations

Reply:
I think at the interface level I feel it is better to have a separate function for BuildDeltaClusters or a generic optionsConfig struct (that defaults for normal cases) instead of forcing every caller to pass some value for delta. Some API design considerations https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

But I would like to hear from @hzxuzhonghu @howardjohn  also.

Reply:
@ramaraochavali addressed

Reply:
Thank you